### PR TITLE
Use a patched LLVM readobj build

### PR DESCRIPTION
**Pull request by jacob-hughes**
LLVM does not currently support parsing of version 3 stackmaps in its `llvm-readobj` tool. In order to use this tool to test our stackmap parsing results against LLVM's, we need to rely on a custom build with patched support for v3 stackmaps. 

Before running `cargo test`, users must ensure that the `LLVM_READOBJ_PATH` environment variable is set to point to the patched `llvm-readobj` binary. They are also required to have `clang` and `llc` in their path. 

The CI for this repo will now use buildbot and bors. Travis has been disabled as it proved too difficult to use to use custom versions of llvm binaries as a dependency.

### CODE REVIEW

**Comment by ltratt** [comment]
bors try

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
I have bors working now, so it's ready for actually reviewing :)

File: .buildbot.sh [hunk 1]
```diff
@@ -1,8 +1,19 @@
...
+RUST_TOOLCHAIN="stable"
+
+set -e 
+
+curl -sSf "https://sh.rustup.rs" | sh -s -- --default-toolchain=$RUST_TOOLCHAIN -y
+export PATH=/opt/llvm-stackmapv3-tools/build/bin/:$HOME/.cargo/bin:$PATH
+export LLVM_READOBJ_PATH=/opt/llvm-stackmapv3-tools/build/bin/llvm-readobj
 cargo test
+
+trap cleanup EXIT
```
Comment (vext01) [inline .buildbot.sh:19]: Does that get called when there's an error too?

File: .buildbot.sh [hunk 2]
```diff
@@ -1,8 +1,19 @@
 #!/bin/sh
-#
-# Build script for continuous integration.
 
-PATH=/opt/rust/bin:${PATH}
-LLVM_READOBJ_PATH=/opt/llvm-stackmapv3-tools/build/bin/llvm-readobj
+#Cleanup code should always happen
```
Comment (vext01) [inline .buildbot.sh]: Space after the hash?

File: .buildbot.sh [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
+cleanup() {
+    rm -rf .rustup/
```
Comment (vext01) [inline .buildbot.sh]: Are you sure that is working? I think the build doesn't begin in ${HOME}, so it's likely these are a no-op?

```
root@bencher12:/home# find buildbot-worker* -name '.rustup' -type d
buildbot-worker1/.rustup
buildbot-worker2/.rustup
buildbot-worker3/.rustup
```

**Review by vext01 — CHANGES_REQUESTED** [review]
A few comments.

File: .buildbot.sh (hunk 3 above)
Comment (jacob-hughes) [reply .buildbot.sh]: No, you're right. This should be `rm -rf $HOME/.rustup/`

Comment (vext01) [reply .buildbot.sh]: Or `~/.rustup`

File: .buildbot.sh (hunk 1 above)
Comment (jacob-hughes) [reply .buildbot.sh:19]: Yes, `cleanup` gets run from any exit from the script

**Comment by jacob-hughes** [comment]
bors try

**Comment by jacob-hughes** [comment]
That seems to have fixed the issue of the rustup dir not being removed. The latest build ran on worker 2 and there's no `.rustup` dir in it's home dir anymore.

**Comment by vext01** [comment]
Good stuff.

Do you want to squash?

P.S.

We should clean the other .rustup dirs in the worker homes.

**Review by vext01 — APPROVED** [review]

**Comment by jacob-hughes** [comment]
Squashed! (Also removed lingering rustup dirs on bencher12 manually)

**Comment by vext01** [comment]
You can now flip the setting that requires bors to pass in the repo settings.

**Pull request merged**